src/main/java/filter/ParseException.java

```diff
@@ -1,8 +1,8 @@
 package filter;
 
 @SuppressWarnings("serial")
-public class ParseException extends RuntimeException {
+public class ParseException extends FilterException {
       public ParseException(String message) {
-         super(message);
+         super("Parse error in filter: " + message);
```

Comment:
Is this going to be shown to the user? In that case, you may want to say 'Error in filter'.